s4e-web/src/app/state/session/session.service.ts

```diff
@@ -1,97 +1,102 @@
+import {InjectorModule} from '../../common/injector.module';
 
 
 @Injectable({providedIn: 'root'})
 export class SessionService {
+  private CONFIG: IConfiguration;
+  private router: Router;
   private _backLink: string;
 
-  constructor(private sessionStore: SessionStore,
-              private sessionQuery: SessionQuery,
-              private _profileQuery: ProfileQuery,
-              private http: HttpClient,
-              private router: Router,
-              private profileService: ProfileService,
-              private CONFIG: S4eConfig) {
+  constructor(private store: SessionStore,
+              private query: SessionQuery,
+              private _notificationService: NotificationService,
+              private http: HttpClient) {
   }
 
   set back_link(back_link: string) {
     this._backLink = back_link;
   }
 
-  init() {
-    const token = localStorage.getItem('token');
-    const email = localStorage.getItem('email') || '';
-
-    if (token != null) {
-      this.sessionStore.update(store => ({...store, token, email}));
-    } else {
-      this.sessionStore.update(store => ({...store, initialized: true, token: null, email: null}));
-    }
+  init(config: IConfiguration) {
+    this.CONFIG = config;
+    this.store.update(store => ({...store, email: null}));
   }
 
-  setToken(token: string | null, email: string | null) {
-    this.sessionStore.update(state => ({...state, token, email}));
-    if (token == null) {
-      localStorage.removeItem('token');
-    } else {
-      localStorage.setItem('token', token);
-    }
+  getProfile$(): Observable<Session | null> {
+    const r = this.http.get<Session>(
+      `${this.CONFIG.apiPrefixV1}/users/me`,
+      {
+        headers: {[ERROR_INTERCEPTOR_SKIP_HEADER]: HTTP_401_UNAUTHORIZED.toString()}
+      }).pipe(
+      tap(profile => this.store.update({...profile})),
+      catchError(() => of(null)),
+      shareReplay(1)
+    );
+
+    r.subscribe();
+
+    return r;
+  }
 
-    if (email == null) {
-      localStorage.removeItem('email');
-    } else {
-      localStorage.setItem('email', email);
-    }
+  resetPassword(oldPassword: string, newPassword: string) {
+    const url = '/api/v1/password-change';
+    return httpPostRequest$(this.http, url, {oldPassword, newPassword}, this.store)
+      .pipe(tap(() => this._notificationService.addGeneral({
+        type: 'success',
+        content: 'Hasło zostało zmienione'
+      })));
   }
 
   @action('login')
   login(formState: LoginFormState) {
-    this.sessionStore.setLoading(true);
+    this.store.setLoading(true);
     const {login: email, password, ...x} = formState;
-    const url = `${this.CONFIG.apiPrefixV1}/token`;
-    this.http.post<LoginRequestResponse>(url, {email, password},
-      {headers: {[ERROR_INTERCEPTOR_CODES_TO_SKIP]: `${HTTP_404_BAD_REQUEST},${HTTP_401_UNAUTHORIZED}`} })
+    const url = `${this.CONFIG.apiPrefixV1}/login`;
+    this.http.post(url, {email, password},
```

Comment:
Why are you not using httpPostRequest$ functionality with 
```
.pipe(        
  switchMap(data => this.getProfile$()),
  tap(() => this.store.update({email: formState.login}))
)
```

Reply:
Because it dos not support custom headers which are being used. It will have to be taken into the consideration when doing general refactoring of http functions